lib/Agrammon/DB/Dataset.pm6

```diff
@@ -11,6 +13,33 @@ class X::Agrammon::DB::Dataset::AlreadyExists is Exception {
     }
 }
 
+#| CSV format error during dataset upload by user.
+class X::Agrammon::DB::Dataset::UploadCSVError is Exception {
+    has Str $.dataset-name is required;
+    has Str $.msg is required;
+    method message {
+        "CSV format error, dataset '$!dataset-name' couldn't be uploaded: $!msg"
+    }
+}
+
+#| Database error during dataset upload by user.
+class X::Agrammon::DB::Dataset::UploadDatabaseFailure is Exception {
+    has Str $.dataset-name is required;
+    has Str $.msg is required;
+    method message {
+        "Database failure, dataset '$!dataset-name' couldn't be uploaded: $!msg"
+    }
+}
+
+#| Database error during dataset upload by user.
```

Comment:
Doc comment is copy-paste of the one above.

Reply:
Fixed.